core/vtk/ttkArrayPreconditioning/ttkArrayPreconditioning.cpp

```diff
@@ -84,6 +83,50 @@ int ttkArrayPreconditioning::RequestData(vtkInformation *ttkNotUsed(request),
     }
   }
 
+  auto vtkGlobalPointIds = pointData->GetGlobalIds();
+  auto vtkGhostCells = pointData->GetArray("vtkGhostType");
+  auto rankArray = pointData->GetArray("RankArray");
+  if(vtkGlobalPointIds != nullptr && vtkGhostCells != nullptr
+     && rankArray != nullptr) {
+#ifdef TTK_ENABLE_MPI
```

Comment:
checking if ttk has been built with MPI is not sufficient to trigger some MPI-specific "behavior".
you need to check that the paraview server has also been executed with `mpirun`.
I believe this can be done by retrieving the MPI communicator from `vtkMPIController` (to check with @CharlesGueunet )

Generally speaking, we need to have to preprocessor check first (#ifdef TTK_ENABLE_MPI), but then a dynamic piece of code (e.g. "isRunningWithMPI()") which checks if the above communicator is `nullptr` or not.

In the present case, if the code is not run with `mpirun`, then simply default back to the original array preconditionning (based on `core/base/common/DisambiguateOrder.h`)

Reply:
the function `isRunningWithMPI()` could be added to `ttkMPIUtils`.
thanks!

Reply:
There is the `MPI_Initialized(int *flag)` method which allows to check whether MPI has been initialized yet (either by Paraview executed with `mpirun` or manually). Do you think about two MPI Utils classes then? One in `base/common` for actual MPI usage in the base code and one in `vtk/ttkAlgorithm` for usage in VTK classes? Or do you mean `base/common/MPI.h` with `ttkMPIUtils`?

Reply:
I guess only one in `core/base/common` is sufficient for the moment. thanks!
(I'll probably rename it to `MPI.h` later as it is going to include more than just utils in the near future).

Reply:
Be aware that the base code should not know anything about VTK.

Reply:
`MPI.h` itself also led to problems on my machine because of the MPI header itself being `mpi.h`. I think case sensitivity depends on the file system, therefore I would suggest a unique name.

Reply:
actually @eve-le-guillou is already renaming it for the next PR.